crates/cairo-addons/src/vm/hint_definitions/dict.rs

```diff
@@ -0,0 +1,69 @@
+use std::{cell::RefCell, collections::HashMap, rc::Rc};
+
+use cairo_vm::{
+    hint_processor::{
+        builtin_hint_processor::{
+            dict_manager::DictManager,
+            hint_utils::{get_ptr_from_var_name, insert_value_from_var_name, insert_value_into_ap},
+        },
+        hint_processor_definition::HintReference,
+    },
+    serde::deserialize_program::ApTracking,
+    types::exec_scope::ExecutionScopes,
+    vm::{errors::hint_errors::HintError, vm_core::VirtualMachine},
+    Felt252,
+};
+
+use crate::vm::hints::Hint;
+
+pub const HINTS: &[fn() -> Hint] = &[dict_new_empty, copy_dict_segment];
+
+pub fn dict_new_empty() -> Hint {
+    Hint::new(
+        String::from("dict_new_empty"),
+        |vm: &mut VirtualMachine,
+         exec_scopes: &mut ExecutionScopes,
+         _ids_data: &HashMap<String, HintReference>,
+         _ap_tracking: &ApTracking,
+         _constants: &HashMap<String, Felt252>|
+         -> Result<(), HintError> {
+            //Check if there is a dict manager in scope, create it if there isnt one
+            let base = if let Ok(dict_manager) = exec_scopes.get_dict_manager() {
+                dict_manager.borrow_mut().new_dict(vm, Default::default())?
+            } else {
+                let mut dict_manager = DictManager::new();
+                let base = dict_manager.new_dict(vm, Default::default())?;
+                exec_scopes.insert_value("dict_manager", Rc::new(RefCell::new(dict_manager)));
+                base
+            };
```

Comment:
I really don't like this if or create for the dict manager in all dict related hints. I have put a dict manager from the beginning in the runner so I think that we can skip it

Reply:
The logic here was to be as little differentiated from the original dict_new hint as possible